neo/Wallets/Wallet.cs

```diff
@@ -182,77 +214,143 @@ public Transaction MakeTransaction(IEnumerable<TransactionAttribute> attributes,
-                    if (balance < amount) return null;
-                    if (assetId.Equals(NativeContract.GAS.Hash))
+                    tx.Gas = Math.Max(engine.GasConsumed - ApplicationEngine.GasFree, 0);
+                    if (tx.Gas > 0)
                     {
-                        balance_gas = balance - amount;
-                        if (balance_gas.Sign <= 0) return null;
+                        long d = (long)NativeContract.GAS.Factor;
+                        long remainder = tx.Gas % d;
+                        if (remainder > 0)
+                            tx.Gas += d - remainder;
+                        else if (remainder < 0)
+                            tx.Gas -= remainder;
                     }
                 }
-            byte[] script;
-            using (ScriptBuilder sb = new ScriptBuilder())
-            {
-                foreach (var output in outputs)
+                UInt160[] hashes = tx.GetScriptHashesForVerifying(snapshot);
+                int size = Transaction.HeaderSize + attributes.GetVarSize() + script.GetVarSize() + IO.Helper.GetVarSize(hashes.Length);
+                foreach (UInt160 hash in hashes)
                 {
-                    sb.EmitAppCall(output.AssetId, "transfer", sender, output.ScriptHash, output.Value.Value);
-                    sb.Emit(OpCode.THROWIFNOT);
+                    script = GetAccount(hash)?.Contract?.Script ?? snapshot.Contracts.TryGet(hash)?.Script;
+                    if (script is null) continue;
+                    if (script.IsSignatureContract())
+                    {
+                        size += 66 + script.GetVarSize();
+                        tx.NetworkFee += ApplicationEngine.OpCodePrices[OpCode.PUSHBYTES64] + ApplicationEngine.OpCodePrices[OpCode.PUSHBYTES33] + InteropService.GetPrice(InteropService.Neo_Crypto_CheckSig, null);
+                    }
+                    else if (script.IsMultiSigContract(out int m, out int n))
+                    {
+                        int size_inv = 65 * m;
+                        size += IO.Helper.GetVarSize(size_inv) + size_inv + script.GetVarSize();
+                        tx.NetworkFee += ApplicationEngine.OpCodePrices[OpCode.PUSHBYTES64] * m;
+                        using (ScriptBuilder sb = new ScriptBuilder())
+                            tx.NetworkFee += ApplicationEngine.OpCodePrices[(OpCode)sb.EmitPush(m).ToArray()[0]];
+                        tx.NetworkFee += ApplicationEngine.OpCodePrices[OpCode.PUSHBYTES33] * n;
+                        using (ScriptBuilder sb = new ScriptBuilder())
+                            tx.NetworkFee += ApplicationEngine.OpCodePrices[(OpCode)sb.EmitPush(n).ToArray()[0]];
+                        tx.NetworkFee += InteropService.GetPrice(InteropService.Neo_Crypto_CheckSig, null) * n;
+                    }
+                    else
+                    {
+                        //We can support more contract types in the future.
```

Comment:
```suggestion
                        //We can support more contract types in the future.
                        return null;
```

Reply:
We don't return `null` so that caller can calculate the size by itself.